src/Sentry/Internal/Hub.cs

```diff
@@ -403,9 +408,10 @@ public SentryId CaptureEvent(SentryEvent evt, SentryHint? hint, Action<Scope> co
         }
     }
 
-    public bool CaptureEnvelope(Envelope envelope) => CurrentClient.CaptureEnvelope(envelope);
-
     public SentryId CaptureEvent(SentryEvent evt, Scope? scope = null, SentryHint? hint = null)
+        => CaptureEvent(evt, scope ?? CurrentScope, scope ?? CurrentScope, hint);
+
+    private SentryId CaptureEvent(SentryEvent evt, Scope eventScope, Scope breadcrumbScope, SentryHint? hint)
```

Comment:
Oh man, this is already giving me a headache, `eventScope` vs `breadcrumbScope`?

Reply:
> Oh man, this is already giving me a headache, `eventScope` vs `breadcrumbScope`?

Yeah I didn't like that much either, but I couldn't see an easier way to do it. The only alternative is to duplicate the code that adds the breadcrumb and insert this in the multiple public overloads of this method.

Reply:
What about putting it into a static helper? All it needs to do is modify a scope that's getting passed in, right?
Rather than juggling two different types of scopes during capture.

Reply:
Yeah, that looks a bit tidier 👍🏻